service/worker/parentclosepolicy/processor.go

```diff
@@ -91,22 +91,34 @@ func New(params *BootstrapParams) *Processor {
 // Start starts the scanner
 func (s *Processor) Start() error {
 	svcClient := s.sdkClientFactory.GetSystemClient()
+
 	processorWorker := s.sdkClientFactory.NewWorker(svcClient, processorTaskQueueName, getWorkerOptions(s))
 	processorWorker.RegisterWorkflowWithOptions(ProcessorWorkflow, workflow.RegisterOptions{Name: processorWFTypeName})
-	processorWorker.RegisterActivityWithOptions(ProcessorActivity, activity.RegisterOptions{Name: processorActivityName})
+	if err := processorWorker.Start(); err != nil {
+		return err
+	}
+
+	activityWorker := s.sdkClientFactory.NewWorker(svcClient, processorActivityTaskQueueName, getActivityWorkerOptions(s))
+	activityWorker.RegisterActivityWithOptions(ProcessorActivity, activity.RegisterOptions{Name: processorActivityName})
 
-	return processorWorker.Start()
+	return activityWorker.Start()
```

Comment:
if activityWorker failed to start, the workflow worker should be stopped I think.